### DISCUSSION TITLE
Anzeige "Ladestand" mit Lücken (Fronius/BYD)

### DISCUSSION THREAD

**Discussion by WolfImBusch** [solectrus/solectrus#3300] [Q&A]
Hallo,

ich verwende Solecturs in Verbindung mit dem Smartmeter Fronius Smartmeter TS 65A-3, Fronius Wechselrichter Fronius Symo Gen24 6.0 Plus sowie einem Stromspeicher BYD Battery-Box Premium HVS.

Bei der Anzeige "Ladestand" sind stets Lücken zu beobachten. Ich vermute (aber auch hier bin ich zu blöd, dies über die influxdb zu prüfen), dass der Ladestand immer nur dann über mqtt übertragen wird, wenn sich dieser ändert.

Hat jemand von euch schon ein ähnliches Verhalten beobachtet?

Ich danke euch und viele Grüße!!

![Ladestand](https://github.com/user-attachments/assets/25a03557-4aef-421b-b7a9-acc23aadf85b)

**Comment by ledermann** [comment]
Ja, das könnte genau die vermutete Ursache habe: Dein MQTT-Broker sendet möglicherweise Daten nur, wenn sich diese ändern. Das solltest du ändern.

Ich benutze für einige Dinge den in ioBroker integrierten MQTT-Broker und habe den so eingestellt:

![CleanShot 2024-07-26 at 17 49 23@2x](https://github.com/user-attachments/assets/2dcda6ba-6ee2-4323-85ba-600657225a36)

Ziel muss sein, dass permanent Werte kommen, auch wenn diese 0 sind oder sich nicht ändern.

**Comment by WolfImBusch** [comment]
Hallo @ledermann,

danke für Deine super-schnelle Antwort und den Hinweis, das Übel an der Quelle anzugehen. In meinem Fall ist das leider nicht der ioBroker, sondern Home Assistant. 

Genauer: ich sammle dort die Daten meiner PV-Komponenten über die "Fronius" Integration erstmal ein. Für Solectrus, d.h. die Werte dann über MQTT zu versenden, verwende ich die im Home Assistant beinhaltete Funktion MQTT Statestream
https://www.home-assistant.io/integrations/mqtt_statestream/
welche über die "configuration.yaml" aktiviert werden kann. 

Leider habe ich dort keine Möglichkeit gefunden, dass Werte zyklisch/ohne Änderung des Wertes versendet werden - also sowas wie "retain" in MQTT Sprache. Mein erster Versuch, die Entität über eine Automatisierung in Home Assistant alle 30s zu aktualisieren in der Hoffnung, dass diese dann verschickt wird, hat leider nicht zum Ziel geführt ...

Hast Du oder ein anderer geneigter Leser, der vielleicht auch Home Assistant verwendet, vielleicht noch eine andere Idee?
Ansonsten auch nicht tragisch - die tolle Optik von Solectrus leidet bei mir nur etwas ... :-)

Herzliche Dank und viele Grüße

**Reply by ledermann** [reply]
Ich bin jetzt nicht so bewandert beim HomeAssistant, aber "MQTT Statestream" ist kein MQTT-Broker. Du verwendest vermutlich "Mosquitto": https://github.com/home-assistant/addons/blob/master/mosquitto/DOCS.md

Ich bin auch interessiert, wie das andere User mit HomeAssistant lösen.

Noch etwas zu "retain": Dieses Flag bedeutet, dass ein Broker den letzten Stand speichert und ihn an einen Client sendet, wenn dieser sich neu verbindet, damit er sofort up-to-date ist. Das ist hier aber nicht relevant. Der MQTT-Collector von SOLECTRUS verbindet sich dauerhaft mit dem Broker, da ist das "retain" nicht entscheidend.

SOLECTRUS benötigt regelmäßig einen aktuellen Wert (auch wenn er sich nicht ändert), um möglichst genaue Summen bilden zu können. Beim Ladestand ist das nicht so kritisch (da geht es wirklich nur um die Lücken im Diagramm), aber bei der Leistung (in Watt) ist es wichtig, dass der Wert regelmäßig aktualisiert wird (auch wenn 0 oder unverändert). Andernfalls wird die Energie (in Watt-Stunden) nicht korrekt berechnet.

Wie sehen denn die anderen Werte bei dir aus? Enthalten die Diagramme für Erzeugung, Einspeisung, Hausverbrauch etc. ebenfalls (größere) Lücken? Das würde Probleme bereiten.

**Reply by WolfImBusch** [reply]
Hallo @ledermann ,

Du hast natürlich recht - jetzt sieht jeder mein gefährliches und in diesem Fall falsches Halbwissen ... :-) Tatsächlich verwende ich den "Mosquitto". Allerdings habe ich diesen nicht über Home Assistant installiert, sondern "separat" als Docker auf meiner Synology NAS, weil ich diesen auch für andere Anwendungen verwende.

Alle Diagramme sind in Ordnung mit Ausnahme des Ladezustands und - wie ich gerade gesehen habe - der Temperatur. Bei letzterem handelt es sich um die Temperatur meiner BYD Batterie. Beide Werte sind dankenswerterweise nicht relevant für die Berechnungen, die Solectrus durchführt.

Aufgefallen ist mir, dass ich das Darstellungsproblem nur bei der Ansicht "TAG" zu haben scheine. Ich habe Dir im Anhang 3 Bilder angehängt ... Spannend ist, dass die Werte dem Anschein nach doch übertragen worden sind - jedoch nicht dargestellt werden. Anhand den Bilder siehst Du sofort, wie ich zu der Vermutung komme ...

Anbei die Ansicht "TAG" für "Ladezustand" und "Temperatur" - dort sind die von mir angesprochene Lücken ...
![IMG_6001](https://github.com/user-attachments/assets/4d1f6fff-5e37-4237-b078-fa499d5128ea)
![IMG_6002](https://github.com/user-attachments/assets/a98f7335-2c9d-49ca-9d90-7e5679ac9677)

Hier noch ein Beispiel "Temperatur" bei der Ansicht "AKTUELL", das in Ordnung zu sein scheint ...
![IMG_6003](https://github.com/user-attachments/assets/3ddc9c1d-8c28-4828-941a-d6797108baa6)

Ich wage es ja kaum zu schreiben, aber könnte es sich nicht doch um ein Darstellungs-"Issue" bei Solectrus handeln?

Gerne kann ich Dir auch die entsprechenden Daten meiner influxDB zur Verfügung stellen - wenn Du mir sagst wie ... :-) Ich glaube mit den Infos aus dieser Rückmeldung kannst Du das Thema wahrscheinlich nicht nachvollziehen. 

Danke Dir und viele Grüße

**Reply by ledermann** [reply]
Ja, du hast recht: Bei JETZT werden etwaige Datenlücken im Diagramm gefüllt (indem der letzte vorhandene Wert herangezogen wird), bei TAG geschieht das nicht. Das ist inkonsequent und ich werde das überdenken.

Grundsätzlich ist es aber so, dass Datenlücken problematisch sind. Nicht bei Temperatur und Ladestand, aber bei der Leistung (Watt). Zur Berechnung der Energie (Wattstunden) mit InfluxDB verwende ich nämlich aus Performancegründe eine Methode, die dann weniger genau ist.

Idealerweise sollt man daher bei Verwendung von MQTT den Broker so einstellen, dass er regelmäßig Werte sendet, auch wenn sich diese nicht geändert haben.

Wenn es bei dir so ist, dass alle Diagramme, die eine **Leistung** anzeigen (Erzeugung, Hausverbrauch, E-Auto etc.), bei TAG **keine** Lücken aufweisen, dann läuft es da korrekt. Achte insbesondere beim E-Auto darauf, dass zu den Zeiten, wo das Auto nicht angeschlossen ist, auch wirklich 0 gesendet wird.
